tests/scripts/verify-operator.sh

```diff
@@ -11,9 +11,39 @@ source ${SCRIPT_DIR}/.definitions.sh
 # Import the check definitions
 source ${SCRIPT_DIR}/checks.sh
 
-check_pod_ready "nvidia-driver-daemonset"
-check_pod_ready "nvidia-container-toolkit-daemonset"
-check_pod_ready "nvidia-device-plugin-daemonset"
-check_pod_ready "nvidia-dcgm-exporter"
-check_pod_ready "gpu-feature-discovery"
-check_pod_ready "nvidia-operator-validator"
+# wait for the nvidia-driver pod to be ready
+# If successful, then wait for the validator pod to be ready (this means that the rest of the pods are healthy)
+# collect log in case of failure
+local start_time=$(date +%s)
+local log_dir=${LOG_DIR}
+while :; do
+    current_time=$(date +%s)
+    elapsed_time=$((current_time - start_time))
+    
+    # Check if total elapsed time is greater than exit
+    if [ $elapsed_time -gt $MAX_POD_STATUS_CHECK_TOTAL_TIME ]; then
+        echo "Total wait time exceeded ${MAX_POD_STATUS_CHECK_TOTAL_TIME} seconds. Exiting..."
```

Comment:
exit when 50 minutes completed

Reply:
Why do we need a while loop and all of this extra logic? Running `kubectl --wait` for both the driver and validator pod should be all we need, and we can exit when a timeout is reached.

Can we simplify this entire file to just:
```
kubectl --wait nvidia-driver-daemonset ...
kubectl --wait nvidia-operator-validator ...

if rc != 0 {
    collect_logs
}
```

And we can extract the `collect_logs` function to its own script.